src/main/kotlin/org/rust/cargo/toolchain/tools/Rustfmt.kt

```diff
@@ -29,49 +36,74 @@ fun RsToolchainBase.rustfmt(): Rustfmt = Rustfmt(this)
 class Rustfmt(toolchain: RsToolchainBase) : RustupComponent(NAME, toolchain) {
 
     fun reformatDocumentTextOrNull(cargoProject: CargoProject, document: Document): String? {
+        val project = cargoProject.project
         return createCommandLine(cargoProject, document)
-            ?.execute(cargoProject.project, stdIn = document.text.toByteArray())
-            ?.unwrapOrElse { if (isUnitTestMode) throw it else return null }
-            ?.stdout
+            ?.execute(project, stdIn = document.text.toByteArray())
+            ?.unwrapOrElse { e ->
+                e.showRustfmtError(project)
+                if (isUnitTestMode) throw e else return null
+            }?.stdout
     }
 
     fun createCommandLine(cargoProject: CargoProject, document: Document): GeneralCommandLine? {
         val file = document.virtualFile ?: return null
         if (file.isNotRustFile || !file.isValid) return null
 
-        val arguments = buildList<String> {
+        val project = cargoProject.project
+        val settings = project.rustfmtSettings
+        val arguments = mutableListOf<String>().apply {
+            if (settings.channel != RustChannel.DEFAULT) add("+${settings.channel}")
+            addAll(ParametersListUtil.parse(settings.additionalArguments))
+
+            removeAll { it.startsWith("--emit") }
```

Comment:
I can pass not only `--emit=value` but also `--emit value`. In this case, this code won't clean up command properly